optuna/structs.py

```diff
@@ -135,6 +135,15 @@ def __ne__(self, other):
 
         return not self.__eq__(other)
 
+    def __lt__(self, other):
+        # type: (Any) -> bool
+
+        if not isinstance(other, type(self)):
```

Comment:
[nits] I mind `type(self)` might be the cause of something problem if we will define subclasses. I think `isinstance(other, FrozenTrial)` is more generic.

Reply:
That makes sense.
Fixed: https://github.com/optuna/optuna/pull/719/commits/c1ae028258d2cd63f00d335ea57ab458325d5293